src/analysis/asymptotic_equivalent.lean

```diff
@@ -0,0 +1,227 @@
+begin
+  rw [is_equivalent, sub_zero],
+  exact is_o_zero_right_iff
+end
+
+lemma is_equivalent_const_iff_tendsto {c : β} (h : c ≠ 0) : u ~[l] (λ _, c) ↔ tendsto u l (𝓝 c) :=
+begin
+  rw [is_equivalent, is_o_const_iff h],
+  split; intro h;
+  [ { have := h.add tendsto_const_nhds, rw zero_add at this },
+    { have := h.add tendsto_const_nhds, rw ← sub_self c} ];
+  convert this; ext; simp [sub_eq_add_neg]
+end
+
+lemma is_equivalent.tendsto_const {c : β} (hu : u ~[l] (λ _, c)) : tendsto u l (𝓝 c) :=
+begin
+  rcases (em $ c = 0) with ⟨rfl, h⟩,
+  { exact (tendsto_congr' $ is_equivalent_zero_iff_eventually_zero.mp hu).mpr tendsto_const_nhds },
+  { exact (is_equivalent_const_iff_tendsto h).mp hu }
+end
+
+end normed_group
+
+open_locale asymptotics
+
+section normed_field
+
+variables {α β : Type*} [normed_field β] {t u v w : α → β} {l : filter α}
+
+lemma is_equivalent_iff_exists_mul_eq : u ~[l] v ↔
+  ∃ (φ : α → β) (hφ : tendsto φ l (𝓝 1)), u =ᶠ[l] φ * v :=
+begin
+  rw [is_equivalent, is_o_iff_exists_eq_mul],
+  split; rintros ⟨φ, hφ, h⟩; [use (φ + 1), use (φ - 1)]; split,
+  { conv in (𝓝 _) { rw ← zero_add (1 : β) },
+    exact hφ.add (tendsto_const_nhds) },
+  { convert h.add (eventually_eq.refl l v); ext; simp [add_mul] },
+  { conv in (𝓝 _) { rw ← sub_self (1 : β) },
+    exact hφ.sub (tendsto_const_nhds) },
+  { convert h.sub (eventually_eq.refl l v); ext; simp [sub_mul] }
+end
+
+lemma is_equivalent.exists_mul_eq (huv : u ~[l] v) :
+  ∃ (φ : α → β) (hφ : tendsto φ l (𝓝 1)), u =ᶠ[l] φ * v :=
+is_equivalent_iff_exists_mul_eq.mp huv
+
+lemma is_equivalent_of_tendsto_one (hz : ∀ᶠ x in l, v x = 0 → u x = 0)
+  (huv : tendsto (u/v) l (𝓝 1)) : u ~[l] v :=
+begin
+  rw is_equivalent_iff_exists_mul_eq,
+  refine ⟨u/v, huv, hz.mono $ λ x hz', (div_mul_cancel_of_imp hz').symm⟩,
+end
+
+lemma is_equivalent_of_tendsto_one' (hz : ∀ x, v x = 0 → u x = 0) (huv : tendsto (u/v) l (𝓝 1)) :
+  u ~[l] v :=
+is_equivalent_of_tendsto_one (eventually_of_forall hz) huv
+
+lemma is_equivalent_iff_tendsto_one (hz : ∀ᶠ x in l, v x ≠ 0) :
+  u ~[l] v ↔ tendsto (u/v) l (𝓝 1) :=
+begin
+  split,
+  { intro hequiv,
+    have := hequiv.is_o.tendsto_0,
+    simp only [pi.sub_apply, sub_div] at this,
+    have key : tendsto (λ x, v x / v x) l (𝓝 1),
+    { exact (tendsto_congr' $ hz.mono $ λ x hnz, @div_self _ _ (v x) hnz).mpr tendsto_const_nhds },
+    convert this.add key,
+    { ext, simp },
+    { norm_num } },
+  { exact is_equivalent_of_tendsto_one (hz.mono $ λ x hnvz hz, (hnvz hz).elim) }
+end
+
+lemma is_equivalent.tendsto_nhds {c : β} (huv : u ~[l] v) (hv : tendsto u l (𝓝 c)) :
+  tendsto v l (𝓝 c) :=
+begin
+  rw ← one_mul c,
+  rcases huv.symm.exists_mul_eq with ⟨φ, hφ, h⟩,
+  exact (tendsto_congr' h.symm).mp (hφ.mul hv)
+end
+
+lemma is_equivalent.tendsto_nhds_iff {c : β} (huv : u ~[l] v) :
+  tendsto u l (𝓝 c) ↔ tendsto v l (𝓝 c) := ⟨huv.tendsto_nhds, huv.symm.tendsto_nhds⟩
+
+lemma is_equivalent.mul (htu : t ~[l] u) (hvw : v ~[l] w) : t * v ~[l] u * w :=
```

Comment:
Does it work for `smul` as well (or more generally for an `is_bounded_bilinear_map`)?

Reply:
Hmmmmmm I need to think about this one a bit more

Reply:
Ok so it won't work for any `is_bounded_bilinear_map` (I have a counterexample with the inner product), but I proved it for smul. Sorry for being so slow, I'm quite busy at the moment :sweat_smile: